doc/src/manual/methods.md

```diff
@@ -389,6 +389,114 @@ false
 The `same_type_numeric` function behaves much like the `same_type` function defined above, but
 is only defined for pairs of numbers.
 
+Redefining Methods
+------------------
+
+When redefining a method or adding new methods,
+it is important to realize that these changes don't take effect immediately.
+This is key to Julia's ability to statically infer and compile code to run fast,
+without the usual JIT tricks and overhead.
+Indeed, any new method definition won't be visible to the current runtime environment,
+including Tasks and Threads (and any previously defined `@generated` functions).
+Let's start with an example to see what this means:
+
+```julia
+julia> function tryeval()
+           @eval newfun() = 1
+           newfun()
+       end
+tryeval (generic function with 1 method)
+
+julia> tryeval()
+ERROR: MethodError: no method matching newfun()
+The applicable method may be too new: running in world age xxxx1, while current world is xxxx2.
+Closest candidates are:
+  newfun() at none:1 (method too new to be called from this world context.)
+ in tryeval() at none:1
+ ...
+
+julia> newfun()
+1
+```
+
+In this example, observe that the new definition for `newfun` has been created,
+but can't be immediately called.
+The new global is immediately visible to the `tryeval` function,
+so you could write `return newfun` (without parentheses).
+But neither you, nor any of your callers, nor the functions they call, or etc.
+can call this new method definition!
+
+But there's an exception: future calls to `newfun` *from the REPL* work as expected,
```

Comment:
I don't really see this as an exception; the REPL isn't a callee of `tryeval` so this is consistent with the story so far.